mint/git-clone/bin/git-clone

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -ueo pipefail
+
+mkdir -p "${CHECKOUT_PATH}"
+cd "${CHECKOUT_PATH}"
+
+if [[ "${PRESERVE_GIT_DIR}" == "true" ]]; then
+  CREDENTIAL_ARG=""
+  if [[ "${CREDENTIAL_HELPER}" != "" ]]; then
+    CREDENTIAL_ARG="-c credential.helper='${CREDENTIAL_HELPER}'"
+  fi
+  eval "git clone $CREDENTIAL_ARG \"${CHECKOUT_REPOSITORY}\" ."
+  git fetch origin "${CHECKOUT_REF}"
+  git checkout "${CHECKOUT_REF}"
+else
+  git init -b main
+  git remote add origin "${CHECKOUT_REPOSITORY}"
+  if [[ "${CREDENTIAL_HELPER}" != "" ]]; then
+    git config --local credential.helper "${CREDENTIAL_HELPER}"
+  fi
+  git fetch --depth=1 origin "${CHECKOUT_REF}"
```

Comment:
Is `--depth=1` the only shallow change?

Reply:
This whole `else` is the change. Previously we used `git clone` when not preserving the git dir. Now we init a new git repo, add an origin, and shallow fetch then checkout the fetch head.

Reply:
Got it, thanks, let me take a closer look then!